internal/pkg/otel/manager/execution_subprocess.go

```diff
@@ -0,0 +1,210 @@
+		return nil, fmt.Errorf("cannot access collector path: %w", err)
+	}
+
+	httpHealthCheckPort, err := findRandomTCPPort()
+	if err != nil {
+		return nil, fmt.Errorf("could not find port for http health check: %w", err)
+	}
+
+	if err := injectHeathCheckV2Extension(cfg, httpHealthCheckPort); err != nil {
+		return nil, fmt.Errorf("failed to inject health check extension: %w", err)
+	}
+
+	confMap := cfg.ToStringMap()
+	confBytes, err := yaml.Marshal(confMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal config to yaml: %w", err)
+	}
+
+	stdOut := runtimeLogger.NewLogWriterWithDefaults(logger.Core(), zapcore.InfoLevel)
+	// info level for stdErr because by default collector writes to stderr
+	stdErr := runtimeLogger.NewLogWriterWithDefaults(logger.Core(), zapcore.InfoLevel)
+
+	procCtx, procCtxCancel := context.WithCancel(ctx)
+	processInfo, err := process.Start(r.collectorPath,
+		process.WithArgs(r.collectorArgs),
+		process.WithContext(procCtx),
+		process.WithEnv(os.Environ()),
+		process.WithCmdOptions(func(c *exec.Cmd) error {
+			c.Stdin = bytes.NewReader(confBytes)
+			c.Stdout = stdOut
+			c.Stderr = stdErr
+			return nil
+		}),
+	)
+	if err != nil {
+		// we failed to start the process
+		procCtxCancel()
+		return nil, fmt.Errorf("failed to start supervised collector: %w", err)
+	}
+	if processInfo.Process == nil {
+		// this should not happen but just in case
+		procCtxCancel()
+		return nil, fmt.Errorf("failed to start supervised collector: process is nil")
+	}
+
+	ctl := &procHandle{
+		processDoneCh: make(chan struct{}),
+		processInfo:   processInfo,
+	}
+
+	healthCheckDone := make(chan struct{})
+	go func() {
+		defer func() {
+			close(healthCheckDone)
+		}()
+		currentStatus := aggregateStatus(componentstatus.StatusStarting, nil)
+		reportStatus(ctx, statusCh, currentStatus)
+		// we will check the health of the collector every 1 second for the first 10 attempts
+		// until we get a successful response.
+		maxFailedAttempts := 10
+		timerDuration := 1 * time.Second
+		healthCheckTicker := time.NewTimer(timerDuration)
+		currentFailedAttempts := 0
+		defer healthCheckTicker.Stop()
+		for {
+			statuses, err := AllComponentsStatuses(procCtx, httpHealthCheckPort)
+			if err != nil {
+				switch {
+				case errors.Is(err, context.Canceled):
+					reportStatus(ctx, statusCh, aggregateStatus(componentstatus.StatusStopped, nil))
+					return
+				case currentFailedAttempts > maxFailedAttempts:
+					reportStatus(procCtx, statusCh, aggregateStatus(componentstatus.StatusFatalError, err))
+					return
+				}
+				currentFailedAttempts++
+			} else {
+				if !compareStatuses(currentStatus, statuses) {
+					currentStatus = statuses
+					reportStatus(procCtx, statusCh, statuses)
+				}
+				// after successfully getting the status, reset the failed attempts,
+				// set the timer duration to 30 seconds and max failed attempts to 3
+				currentFailedAttempts = 0
+				timerDuration = 30 * time.Second
```

Comment:
30s is quite slow, with the control protocol we need "at least" 30s between check-ins and the "at least" is important. The CheckinV2 stream is bi-directional so the sub-processes can push updates back to agent as soon as they happen, so they reflect in agent immediately.

You don't have a way to do the same thing except for polling faster, just keeping this as 1s seems reasonable to me unless the increased duration is mitigating some performance impact or something.

The max failed attempts can still be set to the equivalent of 1m30s.

Reply:
Ok until we have some equivalent to the control protocol for the supervised collector, what you say makes sense, the respective changes are here 8e5632bced158883c08df0de0d92b976fa9ca7be

Reply:
Is this actually true that it's 1.5 minutes of failed attempts?

If it times out, that's 5 seconds per timeout 130 times:

```go
func AllComponentsStatuses(ctx context.Context, httpHealthCheckPort int) (*status.AggregateStatus, error) {
	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
	defer cancel()
```

If it fails immediately, it's 130*1s which is 130s or 2m10s.